lib/puppet/settings.rb

```diff
@@ -1478,7 +1493,8 @@ def lookup(name)
     end
 
     def conf
-      @conf ||= if environments = Puppet.lookup(:environments) { nil }
+      environments = Puppet.lookup(:environments) { nil }
+      @conf ||= if environments
```

Comment:
if `@conf` is not nil, then we used to never lookup the `:environments`. I'm thinking it would be better to write this as:

```ruby
unless @conf
  environments = Puppet.lookup(:environments) { nil }
  @conf = environments.get_conf(@environment_name) if environments
end
```